### DISCUSSION TITLE
设备更新ubuntu到22.04后库不能用了，重新下载源码编译后代码mpp_init 调用异常报错

### DISCUSSION THREAD

**Discussion by livesflying** [rockchip-linux/mpp#466] [Q&A] [answered]
mpp[27161]: mpp_buffer: Assertion caller failed at mpp_buffer_group_init:479
mpp[27161]: mpp_rt: NOT found ion allocator
mpp[27161]: mpp_rt: found drm allocator
mpp[27161]: mpp_buffer: Assertion caller failed at mpp_buffer_group_init:479
mpp[27161]: mpp_info: mpp version: 3c5dc7fe author: sayon.chen    2020-11-23 [h265d_syntax]: Fix h265 syntax fill sps rps issue
mpp[27161]: mpp_log: can not found match soc name: rockchip,rk3588-firefly-itx-3588j-mipi101-M101014-BE45-A1 rockchip,rk3588
mpp[27161]: mpp_serivce: open mpp_service to check cmd capability failed
mpp[27161]: hal_h264e_api_v2: vcodec type 00000000 can not find H.264 encoder device
mpp[27161]: hal_h264e_api_v2: Assertion api failed at hal_h264e_init:65
mpp[27161]: mpp_enc_hal: mpp_enc_hal_init hal hal_h264e init failed ret -1
mpp[27161]: mpp_enc_hal: mpp_enc_hal_init could not found coding type 7
mpp[27161]: mpp_enc_v2: mpp_enc_init_v2 could not init enc hal
mpp[27161]: mpp: error found on mpp initialization
mpp_init failed ret -1

**Comment by FumasterLin** [comment, marked as answer]
release分支很久不维护了。请更新到develop分支

**Reply by livesflying** [reply]
您好更新develop 版本编译后所有的test执行没有任何输出，这是我的配置脚本麻烦帮我看下是缺少什么东西吗

orangepi@firefly:~/mpp-develop/build/linux/aarch64$ ./make-Makefiles.bash
-- cmake version 3.22.1
CMake Deprecation Warning at build/linux/aarch64/arm.linux.cross.cmake:2 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.
Call Stack (most recent call first):
  build/linux/aarch64/CMakeFiles/3.22.1/CMakeSystem.cmake:6 (include)
  CMakeLists.txt:30 (project)


CMake Deprecation Warning at CMakeLists.txt:32 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


-- Detected ARMv8 system processor
-- Checking for one of the modules 'pthread'
CMake Deprecation Warning at osal/CMakeLists.txt:2 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


-- compile with drm support
-- Configuring done
-- Generating done
-- Build files have been written to: /home/orangepi/mpp-develop/build/linux/aarch64

**Comment by HermanChen** [comment]
./make-Makefiles.bash 之后再 make -j 一下
运行之前 export mpp_syslog_perror=1 打开下 log，现在默认写到 syslog 里

**Reply by livesflying** [reply]
您好这是增加log之后的打印信息
mpp_info_test
mpp[322037]: mpp_info_test: normal version log:
mpp[322037]: mpp_info: mpp version: unknown mpp version for missing VCS info
mpp[322037]: mpp_info_test: history version log:
mpp[322037]: mpp_info: mpp version history 0:
mpp[322037]: mpp_compat: id| name -- mpp compat info
mpp[322037]: mpp_compat: 0 | increase decoder fbc buffer size
mpp[322037]: mpp_compat: 1 | support encoder async input mode

**Comment by HermanChen** [comment]
这样正常了，跑下 mpi_dec_test 看看，喂点码流跑下

**Reply by livesflying** [reply]
非常感谢，现在已经好了。
